workers/unity/Packages/com.improbable.gdk.tools/.DownloadCoreSdk/Common.cs

```diff
@@ -0,0 +1,101 @@
+            {
+                Console.WriteLine(((Exception) eventArgs.ExceptionObject).Message);
+                Environment.Exit(1);
+            };
+        }
+
+        public static void EnsureDirectoryEmpty(string dir)
+        {
+            if (Directory.Exists(dir))
+            {
+                Directory.Delete(dir, true);
+            }
+
+            Directory.CreateDirectory(dir);
+        }
+
+        public static int RunRedirectedWithExitCode(string command, IEnumerable<string> arguments)
+        {
+            command = Environment.ExpandEnvironmentVariables(command);
+            arguments = arguments.Select(Environment.ExpandEnvironmentVariables).ToArray();
+
+            var startInfo = new ProcessStartInfo(command, string.Join(" ", arguments.ToArray()))
+            {
+                CreateNoWindow = true,
+                RedirectStandardError = true,
+                RedirectStandardOutput = true,
+                RedirectStandardInput = true,
+                UseShellExecute = false
+            };
+
+            try
+            {
+                using (var process = Process.Start(startInfo))
+                {
+                    if (process != null)
+                    {
+                        process.EnableRaisingEvents = true;
+                        process.OutputDataReceived += (sender, e) =>
+                        {
+                            if (!string.IsNullOrEmpty(e.Data))
+                            {
+                                Console.WriteLine("{0}", e.Data);
+                            }
+                        };
+                        process.ErrorDataReceived += (sender, e) =>
+                        {
+                            if (!string.IsNullOrEmpty(e.Data))
+                            {
+                                Console.Error.WriteLine("{0}", e.Data);
+                            }
+                        };
+
+                        // Async print lines of output as they come in.
+                        process.BeginOutputReadLine();
+                        process.BeginErrorReadLine();
+
+                        process.WaitForExit();
+
+                        return process.ExitCode;
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                throw new Exception($"{ex.Message} while running:\n{command}\n\t{string.Join("\n\t", arguments)}");
+            }
+
+            return 1;
+        }
+
+        public static void RunRedirected(string command, IEnumerable<string> arguments)
+        {
+            var exitCode = RunRedirectedWithExitCode(command, arguments);
+            if (exitCode != 0)
+            {
+                throw new Exception(
+                    $"Exit code {exitCode} while running:\n{command}\n\t{string.Join("\n\t", arguments)}");
+            }
+        }
+
+        public static void WriteHeading(string format, params object[] args)
```

Comment:
Is it worth making a generic version of this that accepts a `ConsoleColor` enum?

Reply:
Maybe if we need to print more colours...hopefully this won't live as a standalone project for *too* long.

Reply:
Well it could be part of a shared library between the various tools?

Reply:
Sure, but it's not yet. :)

Reply:
I have... plans :wink: